arctic/store/version_store.py

```diff
@@ -384,10 +395,24 @@ def get_info(self, symbol, as_of=None):
             return handler.get_info(version)
         return {}
 
+    @staticmethod
+    def handler_supports_read_option( handler, option ):
+        options_method = getattr(handler, "read_options", None)
+        if callable(options_method):
+            return option in options_method()
+
+        # If the handler doesn't support interrogation of its read options assume
+        # that it does support this option (i.e. fail-open)
+        return True
+
     def _do_read(self, symbol, version, from_version=None, **kwargs):
         if version.get('deleted'):
             raise NoDataFoundException("No data found for %s in library %s" % (symbol, self._arctic_lib.get_name()))
         handler = self._read_handler(version, symbol)
+        if AVOID_FALLBACK_HANDLERS:
```

Comment:
The body of the if seems only vaguely related to AVOID_FALLBACK_HANDLERS. Would a more general option that toggles more strict kwarg checking be a better approach?